nifi-nar-bundles/nifi-easyrules-bundle/nifi-easyrules-service/src/main/java/org/apache/nifi/rules/engine/EasyRulesEngineService.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.rules.Action;
+import org.apache.nifi.rules.ActionHandler;
+import org.apache.nifi.rules.Rule;
+import org.apache.nifi.rules.RulesFactory;
+import org.jeasy.rules.api.Condition;
+import org.jeasy.rules.api.Facts;
+import org.jeasy.rules.api.RuleListener;
+import org.jeasy.rules.api.Rules;
+import org.jeasy.rules.core.DefaultRulesEngine;
+import org.jeasy.rules.core.RuleBuilder;
+import org.jeasy.rules.mvel.MVELCondition;
+import org.jeasy.rules.spel.SpELCondition;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of RulesEngineService interface
+ *
+ * @see RulesEngineService
+ */
+@CapabilityDescription("Defines and execute the rules stored in NiFi or EasyRules file formats for a given set of facts. Supports " +
+        "rules stored as JSON or YAML.")
+@Tags({ "rules","rules-engine","engine","actions","facts" })
+public class EasyRulesEngineService  extends AbstractControllerService implements RulesEngineService {
+
+    static final AllowableValue YAML = new AllowableValue("YAML", "YAML", "YAML file configuration type.");
+    static final AllowableValue JSON = new AllowableValue("JSON", "JSON", "JSON file configuration type.");
+    static final AllowableValue NIFI = new AllowableValue("NIFI", "NiFi", "NIFI rules formatted file.");
+    static final AllowableValue MVEL = new AllowableValue("MVEL", "Easy Rules MVEL", "Easy Rules File format using MVFLEX Expression Language");
+    static final AllowableValue SPEL = new AllowableValue("SPEL", "Easy Rules SpEL", "Easy Rules File format using Spring Expression Language");
+
+    static final PropertyDescriptor RULES_FILE_PATH = new PropertyDescriptor.Builder()
+            .name("rules-file-path")
+            .displayName("Rules File Path")
+            .description("Rules File Location")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    static final PropertyDescriptor RULES_FILE_TYPE = new PropertyDescriptor.Builder()
+            .name("rules-file-type")
+            .displayName("Rules File Type")
+            .description("File type for rules definition. Supported file types are YAML and JSON")
+            .required(true)
+            .allowableValues(JSON,YAML)
+            .defaultValue(JSON.getValue())
+            .build();
+
+    static final PropertyDescriptor RULES_FILE_FORMAT = new PropertyDescriptor.Builder()
+            .name("rules-file-format")
+            .displayName("Rules File Format")
+            .description("File format for rules. Supported formats are NiFi Rules.")
```

Comment:
We should include all the currently supported formats in the description

Reply:
Good catch here, I forgot to update (when I only had NiFi rules format implemented:))